src/java/org/apache/cassandra/streaming/async/NettyStreamingMessageSender.java

```diff
@@ -93,10 +95,24 @@
     private Channel controlMessageChannel;
 
     // note: this really doesn't need to be a LBQ, just something that's thread safe
-    private Collection<ScheduledFuture<?>> channelKeepAlives = new LinkedBlockingQueue<>();
+    private final Collection<ScheduledFuture<?>> channelKeepAlives = new LinkedBlockingQueue<>();
 
     private final ThreadPoolExecutor fileTransferExecutor;
 
+    /**
+     * A {@link ThreadLocal} used by the threads in {@link #fileTransferExecutor} to stash references to constructed
+     * and connected {@link Channel}s.
+     */
+    private static final FastThreadLocal<Channel> threadLocalChannel = new FastThreadLocal<>();
+
+    /**
+     * A netty channel attribute used to indicate if a channel is currently transferring a file. This is primarily used
```

Comment:
We discussed offline, and I added comments, but I fixed a stupid bug. `KeepAliveTask` was writing to the control channel, not the channel on which it executed.